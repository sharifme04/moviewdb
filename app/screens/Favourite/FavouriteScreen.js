import React from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {
  TouchableOpacity,
  FlatList,
  View,
  SafeAreaView,
  Text,
} from 'react-native';
import {removeFavourite} from '../../actions/favourite';
import EmptyComponent from '../../components/EmptyComponent';
import styles from './Style';

const FavouriteScreen = ({navigation}) => {
  const dispatch = useDispatch();
  const favourite = useSelector((state) => state.favourite?.favouriteMovies);

  const renderItem = ({item}) => (
    <View style={styles.container}>
      <TouchableOpacity style={styles.item}>
        <Text style={styles.title}>{item}</Text>
      </TouchableOpacity>
      <TouchableOpacity onPress={() => dispatch(removeFavourite(item))}>
        <Text style={styles.delete}>X</Text>
      </TouchableOpacity>
    </View>
  );

  return (
    <SafeAreaView>
      <FlatList
        data={favourite}
        renderItem={renderItem}
        keyExtractor={(item) => item.toString()}
        ListEmptyComponent={
          <EmptyComponent
            navigation={navigation}
            title="Please add your favourite or"
          />
        }
      />
    </SafeAreaView>
  );
};

export default FavouriteScreen;
