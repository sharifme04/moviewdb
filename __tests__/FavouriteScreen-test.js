import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import FavouriteScreen from '../app/screens/Favourite/FavouriteScreen';
import renderer from 'react-test-renderer';

const mockStore = configureMockStore([thunk]);
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
const store = mockStore({});

const fakefavorite = ['Ava'];

test('FavouriteScreen', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <FavouriteScreen />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('FavouriteScreen with data ', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <FavouriteScreen favourite={fakefavorite} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('FavouriteScreen with empty array ', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <FavouriteScreen favourite={[]} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
