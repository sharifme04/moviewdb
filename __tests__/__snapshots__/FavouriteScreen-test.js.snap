// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteScreen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView
    ListEmptyComponent={
      <EmptyComponent
        title="Please add your favourite or"
      />
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          Please add your favourite or
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#f77a55",
              "borderRadius": 25,
              "color": "#ffffff",
              "fontSize": 20,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#f77a55",
                "borderRadius": 25,
                "color": "#ffffff",
                "fontSize": 20,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            Back To home
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`FavouriteScreen with data  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView
    ListEmptyComponent={
      <EmptyComponent
        title="Please add your favourite or"
      />
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          Please add your favourite or
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#f77a55",
              "borderRadius": 25,
              "color": "#ffffff",
              "fontSize": 20,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#f77a55",
                "borderRadius": 25,
                "color": "#ffffff",
                "fontSize": 20,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            Back To home
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`FavouriteScreen with empty array  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView
    ListEmptyComponent={
      <EmptyComponent
        title="Please add your favourite or"
      />
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          Please add your favourite or
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#f77a55",
              "borderRadius": 25,
              "color": "#ffffff",
              "fontSize": 20,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#f77a55",
                "borderRadius": 25,
                "color": "#ffffff",
                "fontSize": 20,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            Back To home
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
